### PR TITLE
Schema registry interface improvement

### PR DESCRIPTION
**Pull request by zilm13**
Improves `SchemaProvider` definition for a better dev exp.

### CODE REVIEW

File: ethereum/spec/src/test/java/tech/pegasys/teku/spec/schemas/registry/BaseSchemaProviderTest.java [hunk 1]
```diff
@@ -13,80 +13,175 @@
...
+
+    assertThat(provider.getSupportedMilestones())
+        .containsExactly(PHASE0, ALTAIR, BELLATRIX, CAPELLA);
   }
 
   @Test
-  void shouldGetNonOverlappingVersionMappings() {
-    provider.addMilestoneMapping(PHASE0, ALTAIR);
-    provider.addMilestoneMapping(BELLATRIX, CAPELLA);
+  void shouldAllowSingleMilestone() {
```
Comment (zilm13) [inline ethereum/spec/src/test/java/tech/pegasys/teku/spec/schemas/registry/BaseSchemaProviderTest.java]: should update test name. Probably is covered by `shouldThrowWhenAskingForAnUnsupportedMilestone`

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/schemas/registry/BaseSchemaProvider.java [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+      schemaProviderCreators.add(new SchemaProviderCreator<>(milestone, creationSchema));
+      return this;
+    }
+
+    public Builder<T> until(final SpecMilestone untilMilestone) {
+      this.untilMilestone = untilMilestone;
+      return this;
+    }
+
+    public Builder<T> constant(final boolean isConstant) {
```
Comment (zilm13) [inline ethereum/spec/src/main/java/tech/pegasys/teku/spec/schemas/registry/BaseSchemaProvider.java]: Nit: I'd expose 2 non-param lazy shortcuts instead, `constant()` and `nonConstant()`, or just single `constant()` as it's useless to call it 2 times and default is nonConstant

**Review by zilm13 — COMMENTED** [review]
LGTM

**Review by tbenr — APPROVED** [review]
LGTM

**Pull request merged**